extensions/wikia/Email/EmailController.class.php

```diff
@@ -396,6 +406,80 @@ protected function getSocialMessages() {
 		];
 	}
 
+	private function generateMobileApplicationsBadges() {
+		$mobileApplicationsLinks = $this->generateMobileApplicationsLinks();
+		$hasMobileApplicationBadges = count( $mobileApplicationsLinks ) > 0;
+		$badges = [
+			'hasMobileApplicationBadges' => $hasMobileApplicationBadges
+		];
+
+		if ( $hasMobileApplicationBadges ) {
+			if ( $mobileApplicationsLinks['android'] ) {
+				$badges['android'] = [
+					'link' => $mobileApplicationsLinks['android'],
+					'src' => ImageHelper::getFileUrl(
+						$this->getMessage( 'emailext-mobile-application-android-badge' )->text() )
+				];
+			}
+			if ( $mobileApplicationsLinks['ios'] ) {
+				$badges['ios'] = [
+					'link' => $mobileApplicationsLinks['ios'],
+					'src' => ImageHelper::getFileUrl(
+						$this->getMessage( 'emailext-mobile-application-ios-badge' )->text() )
+				];
+			}
+		}
+
+		return $badges;
+	}
+
+	private function generateMobileApplicationsLinks() {
+		$mobileApplicationsLinks = [];
+
+		$response = $this->fetchMobileApplicationsDetails();
+		$siteId = $this->wg->CityId;
+
+		if ( $this->applicationsExistFor($siteId, $response ) ) {
+			$mobileApplications = json_decode($response, true);
+
+			foreach ( $mobileApplications['apps'] as &$app ) {
+				foreach ( $app['languages'] as &$language ) {
+					if ( $language['wikia_id'] == $siteId ) {
+						if ( $app['android_release'] ) {
+							$release = $app['android_release'];
+							$mobileApplicationsLinks['android'] = "https://play.google.com/store/apps/details?id=$release";
+						}
+						if ( $app['ios_release'] ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)